python/hail/experimental/phase_by_transmission.py

```diff
@@ -0,0 +1,292 @@
+        It is returned as an ordered array where the first vector corresponds to the first allele,
+        and the second vector (only present if het) the second allele.
+
+        :param CallExpression call: genotype
+        :param ArrayExpression alleles: Alleles at the site
+        :return: Array of one-hot-encoded alleles
+        :rtype: ArrayExpression
+        """
+        return hl.cond(
+            call.is_het(),
+            hl.array([
+                hl.call(call[0]).one_hot_alleles(alleles),
+                hl.call(call[1]).one_hot_alleles(alleles),
+            ]),
+            hl.array([hl.call(call[0]).one_hot_alleles(alleles)])
+        )
+
+    def phase_parent_call(call: hl.expr.CallExpression, transmitted_allele_index: int):
+        """
+        Given a genotype and which allele was transmitted to the offspring, returns the parent phased genotype.
+
+        :param CallExpression call: Parent genotype
+        :param int transmitted_allele_index: index of transmitted allele (0 or 1)
+        :return: Phased parent genotype
+        :rtype: CallExpression
+        """
+        return hl.call(
+            call[transmitted_allele_index],
+            call[hl.int(transmitted_allele_index == 0)],
+            phased=True
+        )
+
+    def phase_diploid_proband(
+            locus: hl.expr.LocusExpression,
+            alleles: hl.expr.ArrayExpression,
+            proband_call: hl.expr.CallExpression,
+            father_call: hl.expr.CallExpression,
+            mother_call: hl.expr.CallExpression
+    ) -> hl.expr.ArrayExpression:
+        """
+        Returns phased genotype calls in the case of a diploid proband
+        (autosomes, PAR regions of sex chromosomes or non-PAR regions of a female proband)
+
+        :param LocusExpression locus: Locus in the trio MatrixTable
+        :param ArrayExpression alleles: Alleles in the trio MatrixTable
+        :param CallExpression proband_call: Input proband genotype call
+        :param CallExpression father_call: Input father genotype call
+        :param CallExpression mother_call: Input mother genotype call
+        :return: Array containing: phased proband call, phased father call, phased mother call
+        :rtype: ArrayExpression
+        """
+
+        proband_v = proband_call.one_hot_alleles(alleles)
+        father_v = hl.cond(
+            locus.in_x_nonpar() | locus.in_y_nonpar(),
+            hl.or_missing(father_call.is_haploid(), hl.array([father_call.one_hot_alleles(alleles)])),
+            call_to_one_hot_alleles_array(father_call, alleles)
+        )
+        mother_v = call_to_one_hot_alleles_array(mother_call, alleles)
+
+        combinations = hl.flatmap(
+            lambda f:
+            hl.zip_with_index(mother_v)
+                .filter(lambda m: m[1] + f[1] == proband_v)
+                .map(lambda m: hl.struct(m=m[0], f=f[0])),
+            hl.zip_with_index(father_v)
+        )
+
+        return (
+            hl.cond(
```

Comment:
suggestion (our "suggestion" comments can be dismissed if you don't care / don't agree):

this can be `hl.case().when(hl.is_defined..., hl.array(...).or_missing()`

Reply:
actually, now that I look at it it should be just `or_missing(hl.is_defined..., hl.array())`